core/lib/zksync_core/src/eth_watch/client.rs

```diff
@@ -54,7 +54,7 @@ const TOO_MANY_RESULTS_ALCHEMY: &str = "response size exceeded";
 
 #[derive(Debug)]
 pub struct EthHttpQueryClient {
-    client: Box<dyn EthInterface>,
+    client: Arc<dyn EthInterface>,
```

Comment:
We used to create a client for each consumer of said interface, now we're going to create a single client and share it among all the consumers.

Reply:
why

Reply:
Because that's how the framework works 😅 
You instantiate the resource once for everyone to use, so that other components don't have to know how resources are instantiated. You make the logic more encapsulated and less entangled. Also, you can inject different implementations without having to change N components.

Reply:
got it. thanks!